server/src/main/java/org/jetbrains/bsp/bazel/server/sync/languages/rust/RustWorkspaceResolver.kt

```diff
@@ -0,0 +1,278 @@
+package org.jetbrains.bsp.bazel.server.sync.languages.rust
+
+import ch.epfl.scala.bsp4j.RustDepKindInfo
+import ch.epfl.scala.bsp4j.RustDependency
+import ch.epfl.scala.bsp4j.RustEnvData
+import ch.epfl.scala.bsp4j.RustFeature
+import ch.epfl.scala.bsp4j.RustPackage
+import ch.epfl.scala.bsp4j.RustProcMacroArtifact
+import ch.epfl.scala.bsp4j.RustRawDependency
+import ch.epfl.scala.bsp4j.RustTarget
+import org.jetbrains.bsp.bazel.server.sync.BazelPathsResolver
+import org.jetbrains.bsp.bazel.server.sync.model.Label
+import org.jetbrains.bsp.bazel.server.sync.model.Module
+
+private data class BazelPackageTargetInfo(
+    val packageName: String,
+    val targetName: String
+)
+
+private data class RustTargetModule(
+    val module: Module,
+    val rustModule: RustModule
+)
+
+class RustWorkspaceResolver(val bazelPathsResolver: BazelPathsResolver) {
+    fun rustPackages(rustBspTargets: List<Module>): List<RustPackage> = rustBspTargets
+        .groupBy { resolvePackage(it).packageName }
+        .mapNotNull(::resolveSinglePackage)
+
+    // We need to resolve all dependencies are provide a list of new bazel targets (deps) to be transformed into packages.
+    fun rustDependencies(
+        rustPackages: List<RustPackage>,
+        rustBspTargets: List<Module>
+    ): Pair<List<RustDependency>, List<RustRawDependency>> {
```

Comment:
imo i'd be nicer to create a data class for it instead of a pair

Reply:
Done

Reply:
Did you push tho?

Reply:
I worked on my own branch and didn't prioritize merging it on this branch as I thought no one would take a look at it on Saturday - guess I was wrong :)